src/server/game/Quests/QuestDef.cpp

```diff
@@ -211,17 +211,31 @@ uint32 Quest::XPValue(Player* player) const
         else if (diffFactor > 10)
             diffFactor = 10;
 
-        uint32 xp = diffFactor * xpentry->Exp[_rewardXPDifficulty] / 10;
-        if (xp <= 100)
-            xp = 5 * ((xp + 2) / 5);
-        else if (xp <= 500)
-            xp = 10 * ((xp + 5) / 10);
-        else if (xp <= 1000)
-            xp = 25 * ((xp + 12) / 25);
+        // Calculate level scaled xp
+        uint32 scaledXP = diffFactor * xpentry->Exp[_rewardXPDifficulty] / 10;
+        if (scaledXP <= 100)
+            scaledXP = 5 * ((scaledXP + 2) / 5);
+        else if (scaledXP <= 500)
+            scaledXP = 10 * ((scaledXP + 5) / 10);
+        else if (scaledXP <= 1000)
+            scaledXP = 25 * ((scaledXP + 12) / 25);
         else
-            xp = 50 * ((xp + 25) / 50);
+            scaledXP = 50 * ((scaledXP + 25) / 50);
+
+        // Calculate non-level scaled xp
+        uint32 minScaledXP = xpentry->Exp[_rewardXPDifficulty];
```

Comment:
This is some serious code duplication right there. Get rid of it and do your replacement before all the if branchings (and, as above, put the replacement in an explicit if branch on your config entry).